swupd/helpers_test.go

```diff
@@ -218,3 +200,37 @@ func mustCreateManifests(t *testing.T, ver uint32, minVer bool, format uint, tes
 		t.Fatal(err)
 	}
 }
+
+func checkManifestContains(t *testing.T, testDir, ver, name string, subs []string, expRes bool) {
+	manFpath := filepath.Join(testDir, "www", ver, "Manifest."+name)
+	b, err := ioutil.ReadFile(manFpath)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	for _, sub := range subs {
+		if bytes.Contains(b, []byte(sub)) != expRes {
+			if expRes {
+				t.Errorf("%s/Manifest.%s did not contain expected '%s'", ver, name, sub)
+			} else {
+				t.Errorf("%s/Manifest.%s contained unexpected '%s'", ver, name, sub)
+			}
+		}
+	}
+}
+
+func checkManifestContainsRes(t *testing.T, testDir, ver, name string, res []*regexp.Regexp) {
```

Comment:
As far as the interface is concerned, I am not sure there is a lot of value in being able to pass in multiple expressions. As @cmarcelo points out for checkManifestContains, using a varargs approach probably makes sense here as well if we do want multiple expressions.